test/kernels/test_qkernel.py

```diff
@@ -515,5 +516,140 @@ def test_user_parameters(self):
             self.assertEqual(list(qkclass.user_param_binds.keys()), qkclass.user_parameters)
 
 
+class TestQuantumKernelBatching(QiskitMachineLearningTestCase):
+    """Test QuantumKernel circuit batching
+
+    Creates a class factory that takes the QuantumInstance class,
+    applies a method decorator to the execute function and returns the modified class.
+    The decorator function records the actual number of circuits being passed
+    to execute does not exceed the batch_size requested by the Quantum Kernel.
+    Also checks that the sum of the batch sizes matches the total number of expected
+    circuits.
+    Checks batching with both statevector simulator and QASM simulator.
+    """
+
+    def __init__(self, methodName="runTest"):
+        super().__init__(methodName)
+        self.circuit_counts = []
```

Comment:
It should be possible to put `self.circuit_counts = []` into the setup() method no? Would avoid having to do an init - which is unusual in a unit test - also it would avoid having to code it explicitly to do that in each unit test since setup is designed to run ahead of the test itself being invoked.

Reply:
Yes, good point! Cheers. I've made that change.